storm-core/src/jvm/backtype/storm/scheduler/multitenant/Node.java

```diff
@@ -334,7 +334,7 @@ public static int countTotalSlotsAlive(Collection<Node> nodes) {
   public static final Comparator<Node> FREE_NODE_COMPARATOR_DEC = new Comparator<Node>() {
     @Override
     public int compare(Node o1, Node o2) {
-      return o2.totalSlotsFree() - o1.totalSlotsFree();
+      return o1.totalSlotsUsed() - o2.totalSlotsUsed();
```

Comment:
Yes, this was the fix for spreading components on isolated nodes when certain workers are having trouble coming up.